### PR TITLE
Gene review service tests

### PR DESCRIPTION
**Pull request by calvinlu3**
This fixes https://github.com/oncokb/oncokb-pipeline/issues/349

Ended up using https://github.com/marchaos/jest-mock-extended to help mock classes.

### CODE REVIEW

File: src/main/webapp/app/pages/curation/collapsible/ReviewCollapsible.tsx [hunk 1]
```diff
@@ -92,7 +93,9 @@ export const ReviewCollapsible = (props: IReviewCollapsibleProps) => {
           icon={faCheck}
           color={SUCCESS}
           onClick={() => {
-            props.handleAccept(props.hugoSymbol, [props.baseReviewLevel as ReviewLevel], props.isGermline);
+            props.handleAccept(props.hugoSymbol, [props.baseReviewLevel as ReviewLevel], props.isGermline).catch(error => {
```
Comment (jfkonecn) [inline src/main/webapp/app/pages/curation/collapsible/ReviewCollapsible.tsx:96]: Should the prop ever be something other than a ReviewLevel object?

File: src/main/webapp/app/service/firebase/firebase-gene-review-service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+          'new',
+          new Review(DEFAULT_USERNAME),
+          'uuid',
+        ),
+      ).rejects.toThrow(SentryError);
+    });
+  });
+
+  describe('acceptChanges', () => {
+    beforeEach(() => _beforeEach());
```
Comment (jfkonecn) [inline src/main/webapp/app/service/firebase/firebase-gene-review-service.spec.ts:98]: Couldn't you put this logic at the `describe('Firebase Gene Review Service', () => {` level?

You can also be high brow about it. 🎩 

```ts
beforeEach(_beforeEach);
```

File: src/main/webapp/app/service/firebase/firebase-gene-service.ts [hunk 3]
```diff
@@ -291,7 +292,7 @@ export class FirebaseGeneService {
         hugoSymbol,
         tumor.excludedCancerTypes_uuid,
         !excludedCancerTypesReview.isChangeReverted,
-        false
+        false,
```
Comment (jfkonecn) [inline src/main/webapp/app/service/firebase/firebase-gene-service.ts:295]: Did prettier add this? If so, why wasn't it do that before?

**Review by jfkonecn — CHANGES_REQUESTED** [review]
Nice clean tests!

File: src/main/webapp/app/pages/curation/collapsible/ReviewCollapsible.tsx (hunk 1 above)
Comment (calvinlu3) [reply src/main/webapp/app/pages/curation/collapsible/ReviewCollapsible.tsx:96]: `props.baseReviewLevel` can be MetaReviewLevel or ReviewLevel. We still render the MetaReviewLevel in this component.

File: src/main/webapp/app/service/firebase/firebase-gene-service.ts (hunk 3 above)
Comment (calvinlu3) [reply src/main/webapp/app/service/firebase/firebase-gene-service.ts:295]: Yeah, prettier added that comma after I removed it and git commited. Not sure why it wasn't added before.

File: src/main/webapp/app/service/firebase/firebase-gene-review-service.spec.ts [hunk 4]
```diff
@@ -38,9 +38,9 @@ describe('Firebase Gene Review Service', () => {
     jest.useFakeTimers().setSystemTime(DEFAULT_DATE);
   };
 
-  describe('updateReviewableContent', () => {
-    beforeEach(() => _beforeEach());
+  beforeEach(_beforeEach);
```
Comment (jfkonecn) [inline src/main/webapp/app/service/firebase/firebase-gene-review-service.spec.ts:41]: Is there a need to have the _beforeEach function as a variable if it's only being used once?

**Review by jfkonecn — APPROVED** [review]
I'm okay with you not addressing my comment if you disagree with it.

File: src/main/webapp/app/service/firebase/firebase-gene-review-service.spec.ts (hunk 4 above)
Comment (calvinlu3) [reply src/main/webapp/app/service/firebase/firebase-gene-review-service.spec.ts:41]: I agree that we can get rid of the function

File: src/main/webapp/app/service/firebase/firebase-gene-review-service.ts [hunk 5]
```diff
@@ -79,136 +58,124 @@ export class FirebaseGeneReviewService {
...
-    return this.firebaseRepository.update(getFirebaseGenePath(isGermline, hugoSymbol), updateObject).then(() => {
-      if (isChangeReverted) {
-        this.firebaseRepository.delete(`${firebasePath}_review/lastReviewed`);
-      }
-      // Update Meta information
-      this.firebaseMetaService.updateGeneMetaContent(hugoSymbol, isGermline);
-      this.firebaseMetaService.updateGeneReviewUuid(hugoSymbol, uuid, !isChangeReverted, isGermline);
-    });
+    try {
+      await this.firebaseMetaService.updateMeta(hugoSymbol, uuid, !isChangeReverted, isGermline);
```
Comment (bprize15) [inline src/main/webapp/app/service/firebase/firebase-gene-review-service.ts:62]: Assuming we didn't come up with a solution for if one of these updates failed then? I'm fine with leaving this, but if you think it's really critical to check both you can look at `createGene` in `firebase-gene-service.ts`. I created a utility method `isPromiseOk` for dealing with this sort of scenario.

File: src/main/webapp/app/service/firebase/firebase-gene-review-service.ts [hunk 6]
```diff
@@ -79,136 +58,124 @@ export class FirebaseGeneReviewService {
...
+      const { uuid, reviewAction, review, reviewPath } = reviewLevel.reviewInfo;
+
+      if (reviewAction === ReviewAction.UPDATE || reviewAction === ReviewAction.NAME_CHANGE) {
+        if (reviewLevel.nestedUnderCreateOrDelete) {
+          continue;
+        }
+        clearReview(review);
+        try {
+          await this.firebaseRepository.update(geneFirebasePath, { [reviewPath]: review });
+          await this.firebaseMetaService.updateGeneReviewUuid(hugoSymbol, uuid, false, isGermline);
```
Comment (bprize15) [inline src/main/webapp/app/service/firebase/firebase-gene-review-service.ts:90]: Same comment as above

File: src/main/webapp/app/oncokb-commons/components/util/NotificationUtils.tsx [hunk 7]
```diff
@@ -36,3 +38,15 @@ export const notifyWarning = (error: OncoKBError, additionalInfo?: string, optio
 export const notifyError = (error: Error | OncoKBError, additionalInfo?: string, options?: ToastOptions) => {
   return toast.error(getErrorMessage(error, additionalInfo), options);
 };
+
+export const notifyCurationPageError = (error: Error | OncoKBError | SentryError, additionalInfo?: string) => {
```
[the remark this answers is not in the document]
Comment (calvinlu3) [reply src/main/webapp/app/oncokb-commons/components/util/NotificationUtils.tsx:42]: Merged into `notifyError`

File: src/main/webapp/app/service/firebase/firebase-gene-review-service.ts (hunk 5 above)
Comment (calvinlu3) [reply src/main/webapp/app/service/firebase/firebase-gene-review-service.ts:62]: I took a look at using multi-location updates (atomic operation), however it does not work well when the update depends on the data that is currently in firebase. For instance, when we accept a deletion, we need to know the current index. This use case is what `runTransaction` is used for, but in order to update both Meta and Gene collections, we need to provide the root path. `runTransaction` also reads in the data given the path, so in our case, we would be reading in our entire firebase database.

I am going to use SentryError to track errors for now.

**Review by bprize15 — APPROVED** [review]
LGTM, thanks for changes!

**Pull request merged**